wapi/auth.py

```diff
@@ -60,7 +58,7 @@ def _authenticate(self):
         self.token_type = rsp['token_type']
         self.valid_until = now + int(rsp['expires_in'] * 0.95)
 
-    def get_headers(self, data):
+    def get_headers(self):
```

Comment:
This is a breaking change if the public `get_headers` is called in client code for some reason. This should then be a major version update.

Reply:
As discussed, I will revert this change and include it in another PR that should be prepared for a major release. This PR's scope will then stop at fixing the multithreading issue.